scripts/find-links-localize.py

```diff
@@ -0,0 +1,154 @@
+                for p in path_output:
+                    f.write(p)
+                    #count = count + 1
+    except OSError as ose:
+        print("[Error] Unable to create temp results file {}; error: {}"
+              .format(result_file, ose))
+        return -2
+    except IOError as e:
+        print(e)
+
+    #print(count)
+
+def find_site_links(content_dir, lang, sub_path):
+    """Find and validate intra-site links on a content page.
+       Look for all en links in localized page.
+       Links such as [...](/docs/...)
+       Check if a localized version exists
+       Replace the en version with the localized file
+       Report en links that are invalid
+       Investigating: Report localized links that are invalid
+    """
+
+    page_and_links = []
+
+    try:
+        with open(sub_path, "r") as mdFile:
+            content = mdFile.read()
+    except Exception as ex:
+        print("[Error] failed in reading markdown file: ".format(ex))
+
+    # Single results: searches for pattern: ](docs paths)
+    # remote_regex = re.compile(r"\]\(\/docs[\/\w\-\#\/]+\)")
+
+    # Single results: searches for pattern: []()
+    remote_regex = re.compile(r"(\[[\w\s\.-]*\])(\(\/docs[\/\w\-\#\/]+\))")
+    matches = remote_regex.findall(content)
+
+    for m in matches:
+        page_and_links.append("\nMatch link:" + m[0] + m[1])
+        # clean up path and split
+        subpaths = m[1][1:-1].split("/")
+
+        # create path to file under localized content directory
+        locale_file = content_dir + "/" + lang
+        for p in subpaths:
+            locale_file = os.path.realpath(os.path.join(locale_file, p)) 
+
+        # some paths may have intra page anchors
+        # strip of #<path-string>, check for page
+        # TODO: check for heading on localized page
+        # default adds .md to path
+        if locale_file.find("#") >= 0:
+            locale_file = locale_file.split("#")[0]
+
+        #check if localized version of file exists
+        locale_file = locale_file + ".md"
+        page_and_links.append("\nLooking for localized file:" + locale_file + "\n")
+
+        if Path(locale_file).is_file():
+            page_and_links.append("\nLocalized file exists, replace EN version with: \n")
+            page_and_links.append(locale_file + "\n")
+        else:
+            # Validate en link
+            locale_file = content_dir + "/en"
+
+            for p in subpaths:
+                locale_file = os.path.realpath(os.path.join(locale_file, p)) 
+
+            if locale_file.find("#") >= 0:
+                locale_file = locale_file.split("#")[0]
+
+            locale_file = locale_file + ".md"
+            if Path(locale_file).is_file():
+                page_and_links.append("\nVALID en link: \n")
+                page_and_links.append(locale_file)
+            else:
+                page_and_links.append("\nNo localized file and INVALID en link:\n")
+                page_and_links.append(locale_file + "\n")
```

Comment:
Some problems I pointed out when attempted to do this thoroughly in another PR (https://github.com/kubernetes/website/pull/19526#issuecomment-600938171).

- You will be treating links to redirect entries as invalid entry, e.g. `/docs/user-guide/`.
- You will have to find out whether `/docs/concepts/overview/` is a valid link, where `overview` is a directory and link target is `/docs/concepts/overview/_index.md`.
- Need to figure out that `/docs/concepts/#kubernetes-control-plane` actually points to `/docs/concepts/_index.md#kubernetes-control-plane`.
- Need to handle link target that contain hugoshort codes
- Need to handle out-of-band link targers such a `[kubenet][kubenet]`

Reply:
@tengqm , thanks for the review. 
Yes, there will be issues with links that have redirects as well as the cases that you note.
I am thinking about checking the _redirect file for links and excluding link targets that contain shortcodes.
What are "out-of-band" link targets, [kubenet][kubenet]?
The links that include in-page anchors will be more complicated to handle.
Would like to start with one main problem (typical links that are invalid and/or have _redirects). Generally, links
to <directory>/_index.md do not change that often (are most of these "index" pages translated).
Is it worthwhile to create a tool to investigate invalid links in .md files?

Reply:
@kbhawkey I believe testing tools can help diagnose quite some problems, and they are in general valuable for identifying nits that skip human eyes. I'm not against the checking or the tools. I just wanted to list some not so uncommon cases for the tools to handle. At the end of the day, the tool may be still very useful if it helps avoid some (even if not all) kinds of link errors.

Maybe I was using the incorrect term, but when I said "out-of-band" links, I was referring to cases like this:

```
The provisioner can be a [kubenet][kubenet] or a [cni plugin][cni-plugin]...

[kubenet]: https://somewhere/you/can/find/kubenet/details
[cni-plugin]: https://another/url/for/cni/specification/or/docs
```